scripts/commit_classifier.py

```diff
@@ -355,6 +355,31 @@ def load_user(phid):
             )
 
             if self.git_repo_dir:
+                # see local commits
+                cmd = "hg out"
+                res = subprocess.check_output(cmd,shell=True)
+                start = res.find('changeset: ')
+                res = res[start:]
+                res = res.replace(" ","")
+                res = res.replace("\n",",")
+
+                # Finding 2nd occurrence of ':' 
+                substr = ":"
+                occurrence = 2
+                inilist = [i for i in range(0, len(ini_str)) \
+                    if ini_str[i:].startswith(substr)] 
+                start = inilist[occurrence - 1] + 1
+                
+                # Finding 1st occurrence of ',' 
+                substr = ","
+                occurrence = 1
+                inilist = [i for i in range(0, len(ini_str)) 
+                            if ini_str[i:].startswith(substr)] 
+                end = inilist[occurrence - 1]
+
+                # hash of the local commit
+                local_commit_hash = res[start:end]
```

Comment:
You should be able to replace all this by using hg.log instead, taking the last applied revisions. You can use https://github.com/mozilla/bugbug/blob/c8ad32b2d6480f65132208448b21cf3eda3a46c8/bugbug/repository.py#L687.

So, first apply all patches to the local mercurial repository, then after the end of the `for patch in needed_stack:` loop, you can get the applied revisions with `get_revs(hg, f"-{len(stack)}")`.

Reply:
Thanks, will do.
But the first comment of #1506 issue says **_"just fetch the commit from the local Mercurial repository where we already applied the patch."_**
So I am assuming that the patches are already applied to the local hg repo. So I will just need to use get_revs to get the hash of latest revision. Is that correct?

Reply:
The code which is applying the patch is the `for patch in needed_stack:` loop I mentioned above :)